src/UI/SlotArea.cpp

```diff
@@ -1680,30 +1681,73 @@ void cSlotAreaEnchanting::UpdateResult(cPlayer & a_Player)
 
 	if (cItem::IsEnchantable(Item.m_ItemType) && Item.m_Enchantments.IsEmpty())
 	{
+		// Pseudocode found at: https://minecraft.gamepedia.com/Enchanting_mechanics
 		int Bookshelves = std::min(GetBookshelvesCount(*a_Player.GetWorld()), 15);
 
-		auto & Random = GetRandomProvider();
+		// Initialise the PRNG using the player's enchantment seed
+		auto EnchantmentSeed = a_Player.GetEnchantmentSeed();
+		MTRand Random(EnchantmentSeed);
+
+		// Calculate the levels for the offered enchantment options:
 		int Base = (Random.RandInt(1, 8) + (Bookshelves / 2) + Random.RandInt(0, Bookshelves));
-		int TopSlot = std::max(Base / 3, 1);
-		int MiddleSlot = (Base * 2) / 3 + 1;
-		int BottomSlot = std::max(Base, Bookshelves * 2);
+		int OptionLevels[3];
+		OptionLevels[0] = std::max(Base / 3, 1);
+		OptionLevels[1] = (Base * 2) / 3 + 1;
+		OptionLevels[2] = std::max(Base, Bookshelves * 2);
+
+		// Properties set according to: https://wiki.vg/Protocol#Window_Property
+		// Send bits of the seed to the client so it can write a bunch of BS in squiggly letters
+		m_ParentWindow.SetProperty(3, static_cast<short>(EnchantmentSeed & 0xFFFFFFF0), a_Player);
```

Comment:
https://wiki.vg/Protocol#Window_Property says this is the bitmask applied (4 bytes) but it the protocol uses shorts (2 bytes) so I'm a little confused.

Reply:
The whole point of the & operation is just to prevent users from deducting the enchantments.

I would say this is your choice how you mask the seed on sending.

I am also wondering why the byte count does mismatch but hey - bugs became features in the past 😁

Reply:
Yep and casting down from uint32 to int16 already hides half the seed to stop deduction haha.